gapis/replay/builder/builder.go

```diff
@@ -580,17 +585,13 @@ func (b *Builder) Build(ctx context.Context) (protocol.Payload, ResponseDecoder,
 		}
 		if err := i.Encode(vml, w); err != nil {
 			err = fmt.Errorf("Encode %T failed for command with id %v: %v", i, id, err)
-			return protocol.Payload{}, nil, err
+			return gapir.Payload{}, nil, err
 		}
 	}
 
-	payload := protocol.Payload{
-		StackSize:          uint32(512), // TODO: Calculate stack size
-		VolatileMemorySize: uint32(vml.size),
-		Constants:          b.constantMemory.data,
-		Resources:          b.resources,
-		Opcodes:            opcodes.Bytes(),
-	}
+	// TODO: This does not looks good
+	payload := *gapir.NewPayload(
+		uint32(512), uint32(vml.size), b.constantMemory.data, b.resources, opcodes.Bytes())
```

Comment:
Suggestion needed. Is this the correct way to avoid extra copy? Is it fine to make NewPayload return the struct, instead of a pointer to the struct?

Reply:
Why not make `Build` return a `*Payload`?

Reply:
Please move the 512 to a named local variable, and keep the TODO